pkg/client/presentproof/client.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package presentproof
+
+import (
+	"errors"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/presentproof"
+)
+
+var (
+	errEmptyRequestPresentation = errors.New("request presentation message is empty")
+	errEmptyProposePresentation = errors.New("propose presentation message is empty")
+)
+
+// Provider contains dependencies for the protocol and is typically created by using aries.Context()
+type Provider interface {
+	Service(id string) (interface{}, error)
+}
+
+// ProtocolService defines the presentproof service.
+type ProtocolService interface {
+	service.DIDComm
+	Actions() ([]presentproof.Action, error)
+	ActionContinue(piID string, opt presentproof.Opt) error
+	ActionStop(piID string, err error) error
+}
+
+// Client enable access to presentproof API
+type Client struct {
+	service.Event
+	service ProtocolService
+}
+
+// New returns new instance of the presentproof client
+func New(ctx Provider) (*Client, error) {
+	raw, err := ctx.Service(presentproof.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	svc, ok := raw.(ProtocolService)
+	if !ok {
+		return nil, errors.New("cast service to presentproof service failed")
+	}
+
+	return &Client{
+		Event:   svc,
+		service: svc,
+	}, nil
+}
+
+// Actions returns unfinished actions for the async usage
+func (c *Client) Actions() ([]presentproof.Action, error) {
+	return c.service.Actions()
+}
+
+// SendRequestPresentation is used by the Verifier to send a request presentation.
+func (c *Client) SendRequestPresentation(msg *presentproof.RequestPresentation, myDID, theirDID string) error {
+	if msg == nil {
+		return errEmptyRequestPresentation
+	}
+
+	msg.Type = presentproof.RequestPresentationMsgType
+
+	_, err := c.service.HandleInbound(service.NewDIDCommMsgMap(msg), myDID, theirDID)
+
+	return err
+}
+
+// AcceptRequestPresentation is used when the Prover is willing to accept the request presentation.
+// NOTE: For async usage.
+func (c *Client) AcceptRequestPresentation(piID string, msg *presentproof.Presentation) error {
+	return c.service.ActionContinue(piID, WithPresentation(msg))
+}
+
+// NegotiateProposePresentation is used when the Prover wants to negotiate about a request presentation he received.
+// NOTE: For async usage. This function can be used only after receiving a request presentation.
+func (c *Client) NegotiateProposePresentation(piID string, msg *presentproof.ProposePresentation) error {
```

Comment:
Shouldn't this method accept a `RequestPresentation` instead?

Reply:
Done! `NegotiateRequestPresentation` sounds good to me.